<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to symetrics’s documentation! &mdash; symetrics 1.0.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            symetrics
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Welcome to symetrics’s documentation!</a></li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a><ul>
<li><a class="reference internal" href="#api-documentation-for-symetrics-api-module">API Documentation for symetrics_api module</a><ul>
<li><a class="reference internal" href="#symetrics.Symetrics"><code class="docutils literal notranslate"><span class="pre">Symetrics</span></code></a><ul>
<li><a class="reference internal" href="#symetrics.Symetrics.connect_to_database"><code class="docutils literal notranslate"><span class="pre">Symetrics.connect_to_database()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_gnomad_constraints"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_gnomad_constraints()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_gnomad_data"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_gnomad_data()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_prop_score"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_prop_score()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_silva_score"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_silva_score()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_spliceai_score"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_spliceai_score()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_surf_score"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_surf_score()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.get_synvep_score"><code class="docutils literal notranslate"><span class="pre">Symetrics.get_synvep_score()</span></code></a></li>
<li><a class="reference internal" href="#symetrics.Symetrics.liftover"><code class="docutils literal notranslate"><span class="pre">Symetrics.liftover()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">symetrics</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Welcome to symetrics’s documentation!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="welcome-to-symetrics-s-documentation">
<h1>Welcome to symetrics’s documentation!<a class="headerlink" href="#welcome-to-symetrics-s-documentation" title="Permalink to this heading"></a></h1>
<div class="toctree-wrapper compound">
</div>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this heading"></a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
<section id="api-documentation-for-symetrics-api-module">
<h2>API Documentation for symetrics_api module<a class="headerlink" href="#api-documentation-for-symetrics-api-module" title="Permalink to this heading"></a></h2>
<h3>Note: Please download the necessary database and lookup dataset first</h3>
<ul>
  <li>Gnomad Database: https://github.com/KalinNonchev/gnomAD_DB</li>
  <li>Symetrics Lookup files: 10.5281/zenodo.1003981 - for update</li>
</ul>
<dl class="py class">
<dt class="sig sig-object py" id="symetrics.Symetrics">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">symetrics.</span></span><span class="sig-name descname"><span class="pre">Symetrics</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">db</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics" title="Permalink to this definition"></a></dt>
<dd><dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.connect_to_database">
<span class="sig-name descname"><span class="pre">connect_to_database</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.connect_to_database"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.connect_to_database" title="Permalink to this definition"></a></dt>
<dd><p>This initalize the connection of the symetrics database set when an instance of the class is created</p>
<dl class="simple">
<dt>Args:</dt><dd><p>None</p>
</dd>
<dt>Returns:</dt><dd><p>conn: an sqlite3 connection object linked to the symetrics database</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_gnomad_constraints">
<span class="sig-name descname"><span class="pre">get_gnomad_constraints</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_gnomad_constraints"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_gnomad_constraints" title="Permalink to this definition"></a></dt>
<dd><p>Get the constraints from gnomad (synonymous z score, missense z score, loss of function z scores, probability of loss of function intolerance) of a given gene</p>
<dl class="simple">
<dt>Args:</dt><dd><p>gene: A string representing the HGNC Symbol of a gene</p>
</dd>
<dt>Returns:</dt><dd><p>gnomad_data: A dictionary of the synonymous z score, missense z score, loss of function z scores, probability of loss of function intolerance)</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gnomad</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_gnomad_constraints</span><span class="p">(</span><span class="n">gene</span> <span class="o">=</span> <span class="s1">&#39;A1BG&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_gnomad_data">
<span class="sig-name descname"><span class="pre">get_gnomad_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">VariantObject</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_gnomad_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_gnomad_data" title="Permalink to this definition"></a></dt>
<dd><p>Get the gnomad information related to the alleles of the given variant (allele count, allele number and allele frequency)</p>
<dl class="simple">
<dt>Args:</dt><dd><p>variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant</p>
</dd>
<dt>Returns:</dt><dd><p>gnomad_data: A dictionary containing the AC, AN, AF and variant information</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant_hg19</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg19</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant_hg38</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg38</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gnomad_hg19</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_gnomad_data</span><span class="p">(</span><span class="n">variant_hg19</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gnomad_hg38</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_gnomad_data</span><span class="p">(</span><span class="n">variant_hg38</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_prop_score">
<span class="sig-name descname"><span class="pre">get_prop_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">group</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'SYNVEP'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_prop_score"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_prop_score" title="Permalink to this definition"></a></dt>
<dd><p>Get the SYMETRICS score for a given gene abd metrics group. The score was calculated from the pooled z proportion test of different
metrics group with their corresponding threhold:</p>
<blockquote>
<div><ul class="simple">
<li><p>SYNVEP: 0.5</p></li>
<li><p>GERP: 4</p></li>
<li><p>CpG:</p></li>
<li><p>CpG_exon: 1</p></li>
<li><p>RSCU:</p></li>
<li><p>dRSCU:</p></li>
<li><p>SpliceAI: 0.8</p></li>
<li><p>SURF: 0.3</p></li>
</ul>
</div></blockquote>
<dl class="simple">
<dt>Args:</dt><dd><p>gene: A string representing the HGNC Symbol of a gene</p>
</dd>
<dt>Returns:</dt><dd><p>scores: A dictionary returning the pvalues and fdr acquired from the test and the score before and after scaling.</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">score</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_prop_score</span><span class="p">(</span><span class="n">group</span> <span class="o">=</span> <span class="s1">&#39;SYNVEP&#39;</span><span class="p">,</span><span class="n">gene</span> <span class="o">=</span> <span class="s1">&#39;A1BG&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_silva_score">
<span class="sig-name descname"><span class="pre">get_silva_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">VariantObject</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_silva_score"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_silva_score" title="Permalink to this definition"></a></dt>
<dd><p>Get the RSCU, dRSCU, GERP and CpG/CpG_Exon of a given variant (reference: hg19)</p>
<dl class="simple">
<dt>Args:</dt><dd><p>variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant</p>
</dd>
<dt>Returns:</dt><dd><p>silva_scores: A dictionary returning the scores along with the variant information.</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg19</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">silva</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_silva_scores</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_spliceai_score">
<span class="sig-name descname"><span class="pre">get_spliceai_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">VariantObject</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_spliceai_score"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_spliceai_score" title="Permalink to this definition"></a></dt>
<dd><p>Get the SpliceAI a given variant (reference: hg38)
<a class="reference external" href="https://spliceailookup.broadinstitute.org/">https://spliceailookup.broadinstitute.org/</a></p>
<dl class="simple">
<dt>Args:</dt><dd><p>variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant</p>
</dd>
<dt>Returns:</dt><dd><p>spliceai_score: A dictionary returning the scores along with the variant information.</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg38</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">spliceai</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_spliceai_score</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_surf_score">
<span class="sig-name descname"><span class="pre">get_surf_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">VariantObject</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_surf_score"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_surf_score" title="Permalink to this definition"></a></dt>
<dd><p>Get the SURF a given variant (reference: hg38)</p>
<dl class="simple">
<dt>Args:</dt><dd><p>variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant</p>
</dd>
<dt>Returns:</dt><dd><p>surf_scores: A dictionary returning the scores along with the variant information.</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg38</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">surf</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_surf_score</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.get_synvep_score">
<span class="sig-name descname"><span class="pre">get_synvep_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">VariantObject</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.get_synvep_score"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.get_synvep_score" title="Permalink to this definition"></a></dt>
<dd><p>Get the SYNVEP a given variant (reference: hg38/hg19)
<a class="reference external" href="https://services.bromberglab.org/synvep/home">https://services.bromberglab.org/synvep/home</a></p>
<dl class="simple">
<dt>Args:</dt><dd><p>variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant</p>
</dd>
<dt>Returns:</dt><dd><p>synvep_scores: A dictionary returning the scores along with the variant information.</p>
</dd>
</dl>
<p>Examples:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant_hg19</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg19</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant_hg38</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg38</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">synvep_hg19</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_synvep_score</span><span class="p">(</span><span class="n">variant_hg19</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">synvep_hg38</span> <span class="o">=</span> <span class="n">symetrics</span><span class="o">.</span><span class="n">get_synvep_score</span><span class="p">(</span><span class="n">variant_hg38</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="symetrics.Symetrics.liftover">
<span class="sig-name descname"><span class="pre">liftover</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variant</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">VariantObject</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/symetrics/symetrics_api.html#Symetrics.liftover"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#symetrics.Symetrics.liftover" title="Permalink to this definition"></a></dt>
<dd><p>Perform a conversion of the variant position based from their original reference to a target reference. If hg38 is given, it will
be converted to hg19 and otherwise</p>
<dl class="simple">
<dt>Args:</dt><dd><p>variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant</p>
</dd>
<dt>Returns:</dt><dd><p>liftover_variant: A VariantObject instance representing the chromosome, position, reference allele and alternative allele of a variant after liftover</p>
</dd>
</dl>
<p>Exampless:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">symetrics</span> <span class="kn">import</span> <span class="o">*</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">symetrics</span> <span class="o">=</span> <span class="n">Symetrics</span><span class="p">(</span><span class="s1">&#39;symetrics.db&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant_hg19</span> <span class="o">=</span> <span class="n">VariantObject</span><span class="p">(</span><span class="nb">chr</span><span class="o">=</span><span class="s1">&#39;7&#39;</span><span class="p">,</span><span class="n">pos</span><span class="o">=</span><span class="s1">&#39;91763673&#39;</span><span class="p">,</span><span class="n">ref</span><span class="o">=</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="n">alt</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="n">genome</span><span class="o">=</span><span class="n">GenomeReference</span><span class="o">.</span><span class="n">hg19</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">variant_hg38</span> <span class="o">=</span> <span class="n">symetrics_db</span><span class="o">.</span><span class="n">liftover</span><span class="p">(</span><span class="n">variant_hg19</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Linnaeus Bundalian.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>
